apps/portal/controllers/error-pages/error404.html

```diff
@@ -11,14 +27,26 @@
         a{color: #2578ff}
         body{border-top:solid 20px #444;padding:0;margin:0;}
     </style>
-
 </head>
 
 <body>
     <i class="icon-unlink error-icon"></i>
     <h1>Error 404</h1>
     <h3>We can't find what you are looking for.</h3>
-    <h4><a href="/portal">Visit Portal</a></h4>
+    <div id="portal" style="visibility: visible; display: block; ..">
+        <h4><a href="/portal">Go to Portal</a></h4>
+    </div>
+    <script type="text/javascript">
+        var link = document.getElementById('portal');
+        if (window.location.pathname.indexOf("/gadgets/") < 0) {
+            link.style.display = 'block';
+            link.style.visibility = 'visible';
+        }
```

Comment:
Thank you for pointing out. Error pages were refactored and these code segments were removed already.